src/platform_impl/web/web_sys/worker.js

```diff
@@ -0,0 +1,10 @@
+onmessage = event => {
```

Comment:
Hmm, I'm confused as to why this is in JavaScript? Is it not possible to write the worker in Rust? Or is it because it'd require a cumbersome `build.rs` step or similar to first compile the Rust code to WASM, and then embed that WASM into the final application?

(I'm totally fine with this being JS either way, mostly just curious).

Reply:
Unfortunately its currently impossible to spawn a worker without at least some JS.
Even the current plan of the [Shared Everything Threads Proposal](https://github.com/WebAssembly/shared-everything-threads) requires a polyfill written in JS to make it work.

In any case, a lot of functionality currently requires JS anyway, its just that it is something `wasm-bindgen` usually takes care of. However spawning workers unfortunately isn't one of those things.

I was briefly considering waiting until I release my `web-thread` crate, but realized that I would require a custom JS solution for single-threaded Wasm anyway, so this works for both.